### PR TITLE
Add IMEX tutorial

### PR DESCRIPTION
**Pull request by wthrowe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Imex/Protocols/ImexSystem.hpp [hunk 1]
```diff
@@ -17,6 +17,8 @@ namespace imex::protocols {
 ///
 /// For efficiency, the tensors in each sector are required to be
```
Comment (yoonso0-0) [inline src/Evolution/Imex/Protocols/ImexSystem.hpp]: Can you expand this line? e.g. "the list of tags in the `tensors` type alias in each sector are required to be ..""

File: docs/Tutorials/Imex.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+
+The unconditional (linear) stability achievable by implicit integrators allows
+them to solve problems not feasible with explicit methods, but it comes at a
+significant price.  While the explicit update $\eqref{eq:example-Euler}$
+directly gave $y_1$, the implicit update $\eqref{eq:example-backwards-Euler}$
+required the equation to be solved for the updated quantity.  In this example
+that was fairly easy, but, in a real application, finding an analytic solution
+is likely to be difficult, if not impossible.  Implicit integrators are
+therefore usually used with a numerical solve of the update equation.  Even for
+simple systems, this solution adds significantly to the computational cost of
```
Comment (yoonso0-0) [inline docs/Tutorials/Imex.md]: solution -> root-finding procedure

File: docs/Tutorials/Imex.md [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+system can define one or more *implicit sectors*, which are subsets of the
+evolved variables with implicit terms that can be solved independently from one
+another.  (If sectors depend on variables from other sectors, the evolution
+system defines the order in which the implicit updates are applied.)
+
+As an example, for a fluid coupled to neutrinos, the neutrinos obey stiff
+equations under some conditions.  If we consider neutrino flavors to be
+non-interacting, they can each be placed in their own implicit sector and
+solved independently.  The fluid equations will usually be non-stiff, so the
+fluid variables will not be in any sector.
```
Comment (yoonso0-0) [inline docs/Tutorials/Imex.md]: Something like,

```
~ in any sector as the (main) variables that needs to be solved.
(They still might appear as arguments for computing the stiff term).
```

File: docs/Tutorials/Imex.md [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+There is a small complexity in handling non-autonomous (time-dependent) systems
+of equations, such as those for evolutions with control systems.  Some methods
+are defined in a way such that the values of time for the explicit and implicit
+parts of a substep appear to be inconsistent.  SpECTRE always uses the time
+derived from the explicit portion of the method, which is equivalent to
+treating time as an additional evolved variable with a constant explicit
+derivative of 1 and implicit derivative of 0.
+
+SpECTRE supports analytic solution of the implicit-step equation as well as two
```
Comment (yoonso0-0) [inline docs/Tutorials/Imex.md]: supports feeding an analytic solution (if present) of the ~

**Review by yoonso0-0 — COMMENTED** [review]
Thanks for writing all these out

**Comment by yoonso0-0** [comment]
@wthrowe  I'll extend the tutorial to DG-FD executables when I submit that PR, so please remind me if I forget to do so :)

**Comment by wthrowe** [comment]
I did one of the changes as a new commit, since it didn't seem to fit in any of the exiting ones, and the rest as a fixup.

**Review by yoonso0-0 — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**